airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -512,19 +512,12 @@ def _create_ti_state_update_query_and_update_state(
 
         query = update(TI).where(TI.id == ti_id_str)
 
-        # This is slightly inefficient as we deserialize it to then right again serialize it in the sqla
-        # TypeAdapter.
-        next_kwargs = None
-        if ti_patch_payload.next_kwargs:
-            from airflow.serialization.serialized_objects import BaseSerialization
-
-            next_kwargs = BaseSerialization.deserialize(ti_patch_payload.next_kwargs)
-
+        # Store next_kwargs directly (already serialized by worker)
         query = query.values(
             state=TaskInstanceState.DEFERRED,
             trigger_id=trigger_row.id,
             next_method=ti_patch_payload.next_method,
-            next_kwargs=next_kwargs,
+            next_kwargs=ti_patch_payload.next_kwargs,
```

Comment:
We should probably change the type on `next_kwargs` to JsonValue

Reply:
Good call, handled for that and even for trigger_kwargs in a similar way